### PR TITLE
3.0 - Fix issue with memcache and domains starting with letter "u"

### PR DESCRIPTION
**Pull request by dakota**

### CODE REVIEW

File: src/Cache/Engine/MemcachedEngine.php [hunk 1]
```diff
@@ -241,8 +241,8 @@ protected function _setOptions()
      */
     protected function _parseServerString($server)
     {
-        if ($server[0] === 'u') {
-            return [$server, 0];
+        if (strpos($server, 'unix://') === 0) {
+            return array($server, 0);
```
Comment (ceeram) [inline src/Cache/Engine/MemcachedEngine.php]: you can keep short array syntax here

**Comment by ceeram** [comment]
besides the short array syntax LGTM :+1:

**Comment by rchavik** [comment]
nitpick: squash :)

File: src/Cache/Engine/MemcachedEngine.php (hunk 1 above)
Comment (dakota) [inline src/Cache/Engine/MemcachedEngine.php]: Copied the code block from the 2.x fix :) Will quickly correct

**Comment by dakota** [comment]
@rchavik 
image

**Comment by rchavik** [comment]
I expected that :)

File: tests/TestCase/Cache/Engine/MemcachedEngineTest.php [hunk 2]
```diff
@@ -446,6 +446,18 @@ public function testConnectIpv6()
...
     /**
+     * test domain starts with u
+     *
+     * @return void
+     */
+    public function testParseServerStringWithU()
+    {
+        $Memcached = new TestMemcachedEngine();
+        $result = $Memcached->parseServerString('udomain.net:13211');
+        $this->assertEquals(array('udomain.net', '13211'), $result);
```
Comment (ADmad) [inline tests/TestCase/Cache/Engine/MemcachedEngineTest.php]: Our codesniffer needs a sniff to shout warnings for not using short array syntax :)

Comment (ADmad) [inline tests/TestCase/Cache/Engine/MemcachedEngineTest.php]: A sniff for it got added to [phpcs](https://github.com/squizlabs/PHP_CodeSniffer/commit/081f49aa967e2cd7c579442ffa05ef0193c6018a) just hours ago, so just need to wait until it's next release :smile:

**Pull request merged**